src/ddsrt/include/dds/ddsrt/ifaddrs.h

```diff
@@ -36,15 +43,38 @@ struct ddsrt_ifaddrs {
 
 typedef struct ddsrt_ifaddrs ddsrt_ifaddrs_t;
 
+/**
+ * @brief Get the interface addresses (@ref ddsrt_ifaddrs) for specific address families
+ * 
+ * Note that array 'afs' must be terminated with an element 'DDSRT_AF_TERM'.
+ * 
+ * @param[out] ifap interface address pointer
+ * @param[in] afs an array of address families
+ * @return a DDS_RETCODE
+ */
 dds_return_t
 ddsrt_getifaddrs(
   ddsrt_ifaddrs_t **ifap,
   const int *afs);
 
+/**
+ * @brief Free the interface addresses
+ * 
+ * @param[in] ifa the interface addresses to free
```

Comment:
It is not just an arbitrary `ddsrt_ifaddrs_t *`, rather it must be the pointer that was returned by `ddsrt_getifaddrs`.